### PR TITLE
Bug 1147964 - Ignore focus request if there is a new window launching

### PR DESCRIPTION
**Pull request by alivedise**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=25d20ddd88579bc80e8fb82402d05de16473adc6

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=29c04e42764ca25bdab7eae37aa7c437998b74fa

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8fae92cae69052fd1af9da97c3a975b71c61e84a

File: apps/system/js/app_window_factory.js [hunk 1]
```diff
@@ -191,11 +197,26 @@
       } else {
         // homescreenWindowManager already listens webapps-launch and open-app.
         // We don't need to check if the launched app is homescreen.
-        new AppWindow(config);
+        this.forgetLastLaunchingWindow();
+        this._launchingApp = new AppWindow(config);
```
Comment (etiennesegonzac) [inline apps/system/js/app_window_factory.js]: nit: I'd extract those 3 lines in a `this.trackLaunchingApp(config)`

File: apps/system/js/wrapper_factory.js [hunk 2]
```diff
@@ -123,7 +144,11 @@
         config.chrome = {
           scrollable: true
         };
+        this.forgetLastLaunchingWindow();
         app = new AppWindow(config);
+        this._launchingApp = app;
```
Comment (etiennesegonzac) [inline apps/system/js/wrapper_factory.js]: same here

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=90f020bef629a3542b7a8dd3442c307392c7db79

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=69c0bc6344bafaaa70d6f7997cab8f00689b9f60

**Pull request merged**